cache/src/main/java/com/uber/buckcache/datastore/impl/ignite/IgniteConfigurationBuilder.java

```
@@ -28,12 +32,17 @@ public IgniteConfigurationBuilder() {
     igniteConfiguration = new IgniteConfiguration();
   }
 
-  public IgniteConfigurationBuilder addMulticastBasedDiscrovery(String multicastIP, Integer multicastPort, List<String> hostIPs) {
+  public IgniteConfigurationBuilder addMulticastBasedDiscrovery(String multicastIP, Integer multicastPort, List<String> hostIPs, String dnsLookupAddress) {
     TcpDiscoverySpi spi = new TcpDiscoverySpi();
+
     TcpDiscoveryVmIpFinder ipFinder = new TcpDiscoveryMulticastIpFinder();
     ((TcpDiscoveryMulticastIpFinder) ipFinder).setMulticastGroup(multicastIP);
     ((TcpDiscoveryMulticastIpFinder) ipFinder).setMulticastPort(multicastPort);
-    ((TcpDiscoveryMulticastIpFinder) ipFinder).setAddresses(hostIPs);
+    if (!Strings.isNullOrEmpty(dnsLookupAddress)) {
```

If both dnsLookupAddress and hostIps are configured we probably want to throw an exception